app/flatpak-builtins-search.c

```diff
@@ -180,6 +178,18 @@ component_get_flatpak_id (AsComponent *app)
   return flatpak_decomposed_dup_id (app_decomposed);
 }
 
+/* as_component_get_branch() seems to return NULL in practice, so use the
+ * bundle id to get the branch
+ */
+static const char *
+component_get_branch (AsComponent *app)
+{
+  AsBundle *app_bundle = as_component_get_bundle (app, AS_BUNDLE_KIND_FLATPAK);
+  const char *app_ref = as_bundle_get_id (app_bundle);
+  g_autoptr(FlatpakDecomposed) app_decomposed = flatpak_decomposed_new_from_ref (app_ref, NULL);
+  return flatpak_decomposed_get_branch (app_decomposed);
```

Comment:
Again, all this could really use a helper that gets the decomposed from the AsComponent and caches it.